chain/chunks/src/lib.rs

```diff
@@ -780,6 +792,21 @@ impl ShardsManager {
         Ok(Some(self.runtime_adapter.get_part_owner(prev_block_hash, part_ord)?) == self.me)
     }
 
+    // We want to sample some parts that depend on prev_block_hash, and local_random_seed
+    fn should_sample_part(&self, prev_block_hash: &CryptoHash, part_ord: u64) -> Result<bool, Error> {
+        let mut s = DefaultHasher::new();
+        let num_total_parts = self.runtime_adapter.num_total_parts(&prev_block_hash) as u64;
+        self.local_random_seed.hash(&mut s);
+        prev_block_hash.hash(&mut s);
+        for _ in 1..NUM_PARTS_TO_BE_SAMPLED {
+            if part_ord == s.finish() % num_total_parts {
+                return Ok(true)
+            }
+            s.write_i8(0); // Just advance the prng's state forward to get another value.
```

Comment:
you can get the same `part_ord` for both parts we sample, but I think that's fine.

Reply:
True. It can also become a part that is owned by me. Relying on randomness to deter attackers enough of the time.